src/contentScript/backgroundProtocol.ts

```diff
@@ -251,6 +253,21 @@ export function liftContentScript<R extends SerializableResponse>(
   };
 }
 
-if (isContentScript()) {
-  browser.runtime.onMessage.addListener(contentScriptListener);
+function addContentScriptListener(): void {
+  if (isContentScript()) {
```

Comment:
I like early-exit functions that avoid unnecessary indentation
```suggestion
  if (!isContentScript()) {
    throw new Error(
      "addContentScriptListener should only be run from the content script"
    );
  }
```

No else necessary

Reply:
Agree, it also fits more naturally as checking the pre-condition of calling the method (a la design by contract programming)